rom/dev/src/verifier.rs

```diff
@@ -69,6 +85,25 @@ impl<'a> ImageVerificationEnv for &mut RomImageVerificationEnv<'a> {
         self.ecc384.verify(&pub_key, &digest, &sig)
     }
 
+    fn lms_verify(
+        &mut self,
+        digest: &ImageDigest,
+        pub_key: &ImageLmsPublicKey,
+        sig: &ImageLmsSignature,
+    ) -> CaliptraResult<bool> {
+        let mut message = [0u8; SHA384_DIGEST_BYTE_SIZE];
+        for i in 0..digest.len() {
+            message[i * 4..][..4].copy_from_slice(&digest[i].to_be_bytes());
+        }
+        let lms_public_key = parse_public_contents::<SHA192_DIGEST_WORD_SIZE>(pub_key.as_bytes())?;
+        let lms_sig = parse_signature_contents::<
```

Comment:
Why do you need to parse this? Why aren't ImageLmsSignature and LmsSignature the same type?

Converting it to bytes and then back into a structure seems silly to me.

Reply:
LMS driver was implemented by Eric as templatized struct while we need our signatures to be of deterministic size. Hence the conversion.

We will explore if we can change this later.